test/cmdlineTests/hashes/input.sol

```diff
@@ -0,0 +1,17 @@
+// SPDX-License-Identifier: GPL-3.0
+pragma solidity >=0.0;
+
+error fileLevelError(uint z);
+
+contract C {
+    struct S { uint x; }
+
+    event ev(uint y);
+    event anon_ev(uint y) anonymous;
+
+    error err(uint z, uint w);
+
+    function f(S memory) public pure {
+        revert fileLevelError(3);
+    }
+}
```

Comment:
I think it would be a good idea to have a library here with both events/errors used and not used by a contract. Maybe also an interface.

I'd also add a free function and an internal one to make sure they do not appear in output.

Reply:
By nothing. The point is that they should not show up if they are not used.